synthesizer/src/program/instruction/operation/hash.rs

```diff
@@ -272,8 +400,8 @@ impl<N: Network, const VARIANT: u8> Display for HashInstruction<N, VARIANT> {
 impl<N: Network, const VARIANT: u8> FromBytes for HashInstruction<N, VARIANT> {
     /// Reads the operation from a buffer.
     fn read_le<R: Read>(mut reader: R) -> IoResult<Self> {
-        // Read the operand.
-        let operands = vec![Operand::read_le(&mut reader)?];
+        // Read the operands.
+        let operands = vec![Operand::read_le(&mut reader)?; expected_num_operands(VARIANT)];
```

Comment:
Won't this clone the first operand, without reading the second?

https://doc.rust-lang.org/std/macro.vec.html

Reply:
Fixed in [590b357](https://github.com/AleoHQ/snarkVM/pull/1576/commits/590b357b4957f6be090109d1c13da44521571421)